blosc/blosc2.c

```diff
@@ -2241,129 +2253,203 @@ static void t_blosc_do_job(void *ctxt)
 
   ntbytes = 0;                /* only useful for decompression */
 
-  if (compress && !(flags & BLOSC_MEMCPYED)) {
-    /* Compression always has to follow the block order */
-    pthread_mutex_lock(&context->count_mutex);
-    context->thread_nblock++;
-    nblock_ = context->thread_nblock;
-    pthread_mutex_unlock(&context->count_mutex);
-    tblock = nblocks;
-  }
-  else {
-    /* Decompression can happen using any order.  We choose
-      sequential block order on each thread */
+  /* Compression always has to follow the block order */
```

Comment:
This comment, although inherited from previous code, it is actually innacurate.  Better express it as:

`/* Get the next block in the queue */`